src/vdom.rs

```diff
@@ -138,71 +174,50 @@ impl VDom<Node> {
 
     pub(crate) fn step(&mut self, host: &Host, input: VDom<()>) {
         match input.vdom {
-            VDomNode::Text { text: tnew, .. } => match &mut self.vdom {
-                VDomNode::Text {
-                    text: told,
-                    state: node,
-                } => {
-                    if tnew != *told {
-                        node.set_text_content(&tnew);
-                        *told = tnew;
+            VDomNode::Text(mut new) => match self.vdom {
+                VDomNode::Text(mut old) => {
+                    let mut node = old.state;
+                    if new.text != old.text {
+                        node.set_text_content(&new.text);
+                        old.update(Text::new(new.text.clone(), old.state));
```

Comment:
We are creating a new text structure for no good reason and then throwing it away. Consider the update function taking the fields as arguments, so we can do `old.update(new.text.clone())`. In that case the update function should also take care of the `node.set_text_content`

Reply:
yes I agree! this is the right thing to do!